### ISSUE TITLE
KeyError: 'rootSymbol'

### ISSUE THREAD

**Issue by fredrik-johansson** [sphinx-doc/sphinx#2364]
I get the following error when trying to build https://github.com/fredrik-johansson/arb/tree/master/doc/source/ with the latest git checkout of sphinx:

```
# Sphinx version: 1.4a1+
# Python version: 2.7.6 (CPython)
# Docutils version: 0.12 release
# Jinja2 version: 2.8
# Last messages:
#   Running Sphinx v1.4a1+
#   loading pickled environment...
#   done
#   building [mo]: targets for 0 po files that are out of date
#   building [html]: targets for 33 source files that are out of date
#   updating environment:
#   0 added, 1 changed, 0 removed
#   reading sources... [100%] index
# Loaded extensions:
#   sphinx.ext.mathjax (1.4a1+) from /usr/local/lib/python2.7/dist-packages/Sphinx-1.4a1.dev_20160303-py2.7.egg/sphinx/ext/mathjax.pyc
#   alabaster (0.7.7) from /usr/local/lib/python2.7/dist-packages/alabaster/__init__.pyc
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.4a1.dev_20160303-py2.7.egg/sphinx/cmdline.py", line 244, in main
    app.build(opts.force_all, filenames)
  File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.4a1.dev_20160303-py2.7.egg/sphinx/application.py", line 286, in build
    self.builder.build_update()
  File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.4a1.dev_20160303-py2.7.egg/sphinx/builders/__init__.py", line 251, in build_update
    'out of date' % len(to_build))
  File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.4a1.dev_20160303-py2.7.egg/sphinx/builders/__init__.py", line 265, in build
    self.doctreedir, self.app))
  File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.4a1.dev_20160303-py2.7.egg/sphinx/environment.py", line 548, in update
    self._read_serial(docnames, app)
  File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.4a1.dev_20160303-py2.7.egg/sphinx/environment.py", line 567, in _read_serial
    self.clear_doc(docname)
  File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.4a1.dev_20160303-py2.7.egg/sphinx/environment.py", line 291, in clear_doc
    domain.clear_doc(docname)
  File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.4a1.dev_20160303-py2.7.egg/sphinx/domains/cpp.py", line 3990, in clear_doc
    rootSymbol = self.data['rootSymbol']
KeyError: 'rootSymbol'
```

**Comment by fredrik-johansson** [comment]
Sorry for the noise. It went away with "make clean"!

**Comment by jakobandersen** [comment]
Thanks for reminding me about this issue. While it can be circumvented by doing a full rebuild it would be nice to fix the problem anyway.
Unfortunately I haven't had time to find a nice small test case with reproducible steps.

**Comment by tk0miya** [comment]
Thank you for reporting. I fixed it at 04219fa.
It comes from incompatibility of internal database (environment.pickle) between 1.3.x and 1.4.x.
I increments the version of it to let know sphinx about incompatibility.

It will full-build automatically if old internal database found.

**Comment by fredrik-johansson** [comment]
Great! It was not a big deal, but it's always nice when things just work automatically without mysterious errors that you have to work around.
